tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -2882,7 +2882,7 @@ tensorflow::Status ConvertSegmentToGraphDef(
       marker_nodes.insert(node_name);
       auto seg_node = segment_def->add_node();
       tensorflow::NodeDefBuilder builder(node_name, "Identity");
-      auto status = builder.Input(connection.inside_node_name, 0, dtype)
+      auto status = builder.Input(connection.inside_node_name, connection.inside_port, dtype)
```

Comment:
This seems to be the right fix based on what you describe. Would you help to verify that, with only the change in this line but not the others in the PR, it fixes the problem in the graph you mentioned? If so, please revert other changes and keep only this line.